source/connection-troubleshooting.txt

```diff
@@ -0,0 +1,343 @@
+see :manual:`Default MongoDB Port </core/authorization/>`.
+
+Verify the User Is in the Authentication Database
+-------------------------------------------------
+
+Verify the user is in the correct authentication database. For more
+information about the authentication database, see
+:ref:`Authentication Error <node-troubleshooting-connection-admin>`.
+
+Configure Your Firewall
+-----------------------
+
+The firewall needs to have an open port for communicating with the MongoDB
+instance. For more information about configuring the firewall, see
+:ref:`Connection Error <node-troubleshooting-connection-firewall>`.
+
+.. _node-troubleshooting-connection-number-connections:
+
+Check the Number of Connections
+-------------------------------
+
+Each ``MongoClient`` instance supports a maximum number of concurrent open
+connections in its connection pool. You can configure the parameter ``maxPoolSize``
+which defines this limit. The default value is ``100``. If there are already a
+number of open connections equal to ``maxPoolSize``, the server waits until
+a connection becomes available. If this wait time exceeds the ``maxIdleTimeMS``
+value, the driver responds with an error.
+
+For more information about how connection pooling works, see
+:ref:`How Does Connection Pooling Work in the Node Driver? <node-faq-connection-pool>`
+in the FAQ.
+
+Too Many Open Connections
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The driver creates the following error message when it attempts to open a
+connection, but it's reached the maximum number of connections:
+
+.. code-block:: none
+   :copyable: false
+
+   connection refused because too many open connections
+
+The following section describes a method that may help resolve the issue.
+
+Check the Number of Connections
+-------------------------------
+
+If you need to create more open connections, increase ``maxPoolSize``. For more
+information about checking the number of connections, see
+:ref:`Error Sending Message <node-troubleshooting-connection-number-connections>`.
+
+Timeout Error
+~~~~~~~~~~~~~
+
+Sometimes when you send a request through the driver to the server, the request
+times out. When this happens, you might receive an error message
+similar to the following message:
+
+.. code-block:: none
+   :copyable: false
+
+   timed out while checking out a connection from connection pool: context canceled
+
+If you receive this error, try the following action to resolve the
+issue.
+
+Set connectTimeoutMS
+--------------------
+
+To prevent the driver from hanging during connection or to prevent the
+driver from spending too much time attempting to reach unreachable replica set
+nodes, you can set the ``connectTimeoutMS`` option of your
+:ref:`connection options <node-connection-options>`.
+You should ensure the ``connectTimeoutMS`` setting is not lower than
+the highest network latency you have to a member of the set. If one of the
+secondary members is on the other side of the planet and has a latency of 10000
```

Comment:
Suggestion:

I think it's possible to leave out the potential cause for latency since that isn't important to the explanation but might be perceived that way. "If [both] X and Y, then..."

```suggestion
secondary members has a latency of 10000
```